src/inset_state/fill_with_density.cpp

```diff
@@ -8,7 +8,11 @@ void InsetState::fill_with_density(bool plot_density)
   // CartogramInfo::replace_missing_and_zero_target_areas()
   double mean_density = total_target_area() / total_inset_area();
 
+  // Correct mean density if any area drift is present
+  mean_density *= area_drift();
```

Comment:
This problem is not quite so simple. Let `curr_non_na_interior_area` be the current total area of non-NA GeoDivs. The task is to choose the mean density as the ratio of `target_na_and_exterior_area == lx_ * ly_ - initial_non_na_interior_area` to `curr_na_ext_area == lx_ * ly_ - curr_non_na_interior_area`. Consequently:

```
mean_density = (1.0 - initial_non_na_area / (lx_ * ly_)) / (1.0 - current_non_na_area / (lx_ * ly_))
```

Reply:
However, all the other densities are scaled up with respect to "target areas". Thus, would we not need to factor that in somewhere too? We should perhaps discuss this later today once. Further, all the "NA" correction was also done in `replace_missing_and_zero_target_area()`. Thus, could we simplify by not checking for NA?

Reply:
On a small side note, should we also replace `total_inset_area()` with just `area`? The fact that it is the area of the inset is implicit in having it called from an object of the class inset or inside a member function. We could make similar changes in other places such as `set_inset_name` --> `set_name`, and `total_target_area` --> `target_area`.